### DISCUSSION TITLE
Run migrations with FlywayDB - Error: cross-shard query with aggregates

### DISCUSSION THREAD

**Discussion by dbrown428** [planetscale/discussion#159] [Q&A]
Hello,

I'm wondering if anyone has successfully run their migrations using FlywayDB, and if yes, did you have to do anything special for PlanetScale? 

I'm having trouble getting my migrations to run using FlywayDB. I get the following error:
```
ERROR: 
Unable to check whether schema `database` is empty
------------------------------------------------
SQL State  : 42000
Error Code : 1235
Message    : unsupported: cross-shard query with aggregates

Caused by: java.sql.SQLSyntaxErrorException: unsupported: cross-shard query with aggregates

```

I can manually run the SQL code that's in the migration and it works fine (it's a simple CREATE TABLE statement). Does anyone know what a cross-shard query is, and know how to stop FlywayDB from doing it or how to fix it? Or why it's a problem on PlanetScale? I've never experienced this on other DBs.

Here's the SQL I'm trying to run:
```
CREATE TABLE request (
  id binary(16) NOT NULL,
  email varchar(255) NOT NULL,
  created_at timestamp NOT NULL,
  PRIMARY KEY (`id`)
);
```

Any ideas or suggestions would be greatly appreciated.
Thanks a bunch!

**Comment by dbrown428** [comment]
Just putting this info here in case anyone else runs into this issue. I did a little digging and found that the commands FlywayDB tries to run on start do not appear to be supported by Vitess at this time. If you disable `createSchemas` in Flyway configuration and manually create the schema history table, that appears to allow Flyway to run migrations correctly.

An example of a Flyway [SQL statement](https://github.com/flyway/flyway/blob/e4014d71a84f65bb613f9abd5b286c6ac11ff0f8/flyway-mysql/src/main/java/org/flywaydb/database/mysql/MySQLSchema.java#L36) that is not supported at this time:
```
SELECT COUNT(1) FROM information_schema.schemata WHERE schema_name=? LIMIT 1

Result:
Error: ER_NOT_SUPPORTED_YET: Comparing table schema name with a column name not yet supported
```

You can disable auto creation of the flyway history schema in the Flyway configuration `flyway.createSchemas=false`, and then manually create the  [**flyway_history_schema**](https://github.com/flyway/flyway/blob/e4014d71a84f65bb613f9abd5b286c6ac11ff0f8/flyway-mysql/src/main/java/org/flywaydb/database/mysql/MySQLDatabase.java#L160):
```
CREATE TABLE flyway_schema_history (
    installed_rank INT NOT NULL,
    version VARCHAR(50),
    description VARCHAR(200) NOT NULL,
    type VARCHAR(20) NOT NULL,
    script VARCHAR(1000) NOT NULL,
    checksum INT,
    installed_by VARCHAR(100) NOT NULL,
    installed_on TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
    execution_time INT NOT NULL,
    success BOOL NOT NULL,
    CONSTRAINT flyway_schema_history_pk PRIMARY KEY (installed_rank)
);
```

Now you can run your migrations with Flyway.

**Reply by sajidbarcha** [reply]
Are you using Spring Boot, what are constraints exist working with PS compared to a Managed DB (like AWS Postgres RDS) that are specific to PlanetScale (and not specific to MySQL), I am considering usiing PS in my next Spring Boot (rest only) app
